src/CodeGeneration.Roslyn.Engine/CompilationGenerator.cs

```diff
@@ -84,7 +84,7 @@ public class CompilationGenerator
         /// <param name="progress">Optional handler of diagnostics provided by code generator.</param>
         /// <param name="cancellationToken">Cancellation token to interrupt async operations.</param>
         /// <returns>A <see cref="Task.CompletedTask"/>.</returns>
-        public async Task GenerateAsync(IProgress<Diagnostic> progress = null, CancellationToken cancellationToken = default)
+        public async Task GenerateAsync(IProgress<Diagnostic> progress, CancellationToken cancellationToken)
```

Comment:
`CancellationToken` arguments are always last, and always optional. It's a good convention and we should keep it as such.